crates/rattler_package_streaming/src/seek.rs

```diff
@@ -100,19 +121,26 @@ fn get_file_from_archive(
 pub fn read_package_file<P: PackageFile>(path: impl AsRef<Path>) -> Result<P, ExtractError> {
     // stream extract the file from a package
     let file = File::open(&path)?;
+    let content = read_package_file_content::<P>(&file, &path)?;
 
-    match ArchiveType::try_from(&path).ok_or(ExtractError::UnsupportedArchiveType)? {
-        ArchiveType::TarBz2 => {
-            let mut archive = stream_tar_bz2(file);
-            let buf = get_file_from_archive(&mut archive, P::package_path())?;
-            P::from_str(&String::from_utf8_lossy(&buf))
-                .map_err(|e| ExtractError::ArchiveMemberParseError(P::package_path().to_owned(), e))
-        }
-        ArchiveType::Conda => {
-            let mut info_archive = stream_conda_info(file).unwrap();
-            let buf = get_file_from_archive(&mut info_archive, P::package_path())?;
-            P::from_str(&String::from_utf8_lossy(&buf))
-                .map_err(|e| ExtractError::ArchiveMemberParseError(P::package_path().to_owned(), e))
-        }
-    }
+    P::from_str(&String::from_utf8_lossy(&content))
+        .map_err(|e| ExtractError::ArchiveMemberParseError(P::package_path().to_owned(), e))
+}
+
+/// Get a [`PackageFile`] from temporary archive and return it as a temporary file.
+pub fn get_package_file<P: PackageFile>(
+    archive_temp_file: NamedTempFile,
```

Comment:
You can just take the `path` for the NamedTempFile? I am not sure why we need this in the signature?

Reply:
Or let me rephrase this. I think the interface should be slightly more generic. It should be something like:

```
extract_package_file<P: PackageFile, W: Write>(path: &Path, writer: &mut W) -> io::Result<...> {
   ... write the extracted file to the writer ...
}
```